{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4a4268ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Initializing Hail with default parameters...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-09-23 15:48:25 WARN  NativeCodeLoader:60 - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting default log level to \"WARN\".\n",
      "To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-09-23 15:48:26 WARN  Hail:43 - This Hail JAR was compiled for Spark 3.1.1, running with Spark 3.1.2.\n",
      "  Compatibility is not guaranteed.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Running on Apache Spark version 3.1.2\n",
      "SparkUI available at http://c20n06.ruddle.hpc.yale.internal:4040\n",
      "Welcome to\n",
      "     __  __     <>__\n",
      "    / /_/ /__  __/ /\n",
      "   / __  / _ `/ / /\n",
      "  /_/ /_/\\_,_/_/_/   version 0.2.74-0c3a74d12093\n",
      "LOGGING: writing to /gpfs/ycga/scratch60/lek/ml2529/gruber_analysis/hail-20210923-1548-0.2.74-0c3a74d12093.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "Global fields:\n",
      "    None\n",
      "----------------------------------------\n",
      "Column fields:\n",
      "    's': str\n",
      "----------------------------------------\n",
      "Row fields:\n",
      "    'locus': locus<GRCh38>\n",
      "    'alleles': array<str>\n",
      "    'rsid': str\n",
      "    'qual': float64\n",
      "    'filters': set<str>\n",
      "    'info': struct {\n",
      "        AC: array<int32>, \n",
      "        AF: array<float64>, \n",
      "        AN: int32, \n",
      "        BaseQRankSum: float64, \n",
      "        DB: bool, \n",
      "        DP: int32, \n",
      "        END: int32, \n",
      "        ExcessHet: float64, \n",
      "        FS: float64, \n",
      "        InbreedingCoeff: float64, \n",
      "        MLEAC: array<int32>, \n",
      "        MLEAF: array<float64>, \n",
      "        MQ: float64, \n",
      "        MQRankSum: float64, \n",
      "        NEGATIVE_TRAIN_SITE: bool, \n",
      "        POSITIVE_TRAIN_SITE: bool, \n",
      "        QD: float64, \n",
      "        RAW_MQandDP: array<int32>, \n",
      "        ReadPosRankSum: float64, \n",
      "        SOR: float64, \n",
      "        VQSLOD: float64, \n",
      "        culprit: str\n",
      "    }\n",
      "    'allele_data': struct {\n",
      "        nonsplit_alleles: array<str>, \n",
      "        has_star: bool, \n",
      "        variant_type: str, \n",
      "        n_alt_alleles: int32, \n",
      "        allele_type: str, \n",
      "        was_mixed: bool\n",
      "    }\n",
      "    'a_index': int32\n",
      "    'was_split': bool\n",
      "    'vep_proc_id': struct {\n",
      "        part_idx: int32, \n",
      "        block_idx: int32\n",
      "    }\n",
      "    'sortedTranscriptConsequences': array<struct {\n",
      "        biotype: str, \n",
      "        canonical: int32, \n",
      "        cdna_start: int32, \n",
      "        cdna_end: int32, \n",
      "        codons: str, \n",
      "        gene_id: str, \n",
      "        gene_symbol: str, \n",
      "        hgvsc: str, \n",
      "        hgvsp: str, \n",
      "        transcript_id: str, \n",
      "        amino_acids: str, \n",
      "        lof: str, \n",
      "        lof_filter: str, \n",
      "        lof_flags: str, \n",
      "        lof_info: str, \n",
      "        polyphen_prediction: str, \n",
      "        protein_id: str, \n",
      "        protein_start: int32, \n",
      "        sift_prediction: str, \n",
      "        consequence_terms: array<str>, \n",
      "        domains: array<str>, \n",
      "        major_consequence: str, \n",
      "        category: str, \n",
      "        hgvs: str, \n",
      "        major_consequence_rank: int32, \n",
      "        transcript_rank: int32\n",
      "    }>\n",
      "    'gene_symbol': str\n",
      "    'ENSG': str\n",
      "    'ENST': str\n",
      "    'major_consequence': str\n",
      "    'hgvsc': str\n",
      "    'hgvs': str\n",
      "    'category': str\n",
      "    'canonical': int32\n",
      "    'polyphen_pred': str\n",
      "    'sift_pred': str\n",
      "    'most_severe_consequence': str\n",
      "    'variant_class': str\n",
      "    'gene_id_array': array<str>\n",
      "    'gene_symbol_array': array<str>\n",
      "    'variant_qc': struct {\n",
      "        dp_stats: struct {\n",
      "            mean: float64, \n",
      "            stdev: float64, \n",
      "            min: float64, \n",
      "            max: float64\n",
      "        }, \n",
      "        gq_stats: struct {\n",
      "            mean: float64, \n",
      "            stdev: float64, \n",
      "            min: float64, \n",
      "            max: float64\n",
      "        }, \n",
      "        AC: array<int32>, \n",
      "        AF: array<float64>, \n",
      "        AN: int32, \n",
      "        homozygote_count: array<int32>, \n",
      "        call_rate: float64, \n",
      "        n_called: int64, \n",
      "        n_not_called: int64, \n",
      "        n_filtered: int64, \n",
      "        n_het: int64, \n",
      "        n_non_ref: int64, \n",
      "        het_freq_hwe: float64, \n",
      "        p_value_hwe: float64\n",
      "    }\n",
      "    'gnomad_genomes_af': float64\n",
      "    'gnomad_exomes_af': float64\n",
      "    'primate_ai': float64\n",
      "    'splice_ai': float32\n",
      "    'meta_svm': str\n",
      "    'cadd': float32\n",
      "    'gnomad_af': float64\n",
      "    'AC_adj': int32\n",
      "----------------------------------------\n",
      "Entry fields:\n",
      "    'AD': array<int32>\n",
      "    'DP': int32\n",
      "    'GQ': int32\n",
      "    'GT': call\n",
      "    'MIN_DP': int32\n",
      "    'PGT': call\n",
      "    'PID': str\n",
      "    'PL': array<int32>\n",
      "    'PS': int32\n",
      "    'RGQ': int32\n",
      "    'SB': array<int32>\n",
      "    'adj': bool\n",
      "----------------------------------------\n",
      "Column key: ['s']\n",
      "Row key: ['locus', 'alleles']\n",
      "----------------------------------------\n"
     ]
    }
   ],
   "source": [
    "import hail as hl\n",
    "from pcgc_hail.hail_scripts.utils import *\n",
    "from pcgc_hail.hail_scripts.annotate_frequencies import *\n",
    "\n",
    "mt = hl.read_matrix_table('/gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/gruber_WES_b5_vep_filtered_FINAL.mt')\n",
    "mt.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "660d2e27",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reads in the following txt as a hail table: \n",
      "sample_groups.txt\n",
      "\n",
      "----------------------------------------\n",
      "Global fields:\n",
      "    None\n",
      "----------------------------------------\n",
      "Column fields:\n",
      "    's': str\n",
      "----------------------------------------\n",
      "Row fields:\n",
      "    'locus': locus<GRCh38>\n",
      "    'alleles': array<str>\n",
      "    'rsid': str\n",
      "    'qual': float64\n",
      "    'filters': set<str>\n",
      "    'info': struct {\n",
      "        AC: array<int32>, \n",
      "        AF: array<float64>, \n",
      "        AN: int32, \n",
      "        BaseQRankSum: float64, \n",
      "        DB: bool, \n",
      "        DP: int32, \n",
      "        END: int32, \n",
      "        ExcessHet: float64, \n",
      "        FS: float64, \n",
      "        InbreedingCoeff: float64, \n",
      "        MLEAC: array<int32>, \n",
      "        MLEAF: array<float64>, \n",
      "        MQ: float64, \n",
      "        MQRankSum: float64, \n",
      "        NEGATIVE_TRAIN_SITE: bool, \n",
      "        POSITIVE_TRAIN_SITE: bool, \n",
      "        QD: float64, \n",
      "        RAW_MQandDP: array<int32>, \n",
      "        ReadPosRankSum: float64, \n",
      "        SOR: float64, \n",
      "        VQSLOD: float64, \n",
      "        culprit: str\n",
      "    }\n",
      "    'allele_data': struct {\n",
      "        nonsplit_alleles: array<str>, \n",
      "        has_star: bool, \n",
      "        variant_type: str, \n",
      "        n_alt_alleles: int32, \n",
      "        allele_type: str, \n",
      "        was_mixed: bool\n",
      "    }\n",
      "    'a_index': int32\n",
      "    'was_split': bool\n",
      "    'vep_proc_id': struct {\n",
      "        part_idx: int32, \n",
      "        block_idx: int32\n",
      "    }\n",
      "    'sortedTranscriptConsequences': array<struct {\n",
      "        biotype: str, \n",
      "        canonical: int32, \n",
      "        cdna_start: int32, \n",
      "        cdna_end: int32, \n",
      "        codons: str, \n",
      "        gene_id: str, \n",
      "        gene_symbol: str, \n",
      "        hgvsc: str, \n",
      "        hgvsp: str, \n",
      "        transcript_id: str, \n",
      "        amino_acids: str, \n",
      "        lof: str, \n",
      "        lof_filter: str, \n",
      "        lof_flags: str, \n",
      "        lof_info: str, \n",
      "        polyphen_prediction: str, \n",
      "        protein_id: str, \n",
      "        protein_start: int32, \n",
      "        sift_prediction: str, \n",
      "        consequence_terms: array<str>, \n",
      "        domains: array<str>, \n",
      "        major_consequence: str, \n",
      "        category: str, \n",
      "        hgvs: str, \n",
      "        major_consequence_rank: int32, \n",
      "        transcript_rank: int32\n",
      "    }>\n",
      "    'gene_symbol': str\n",
      "    'ENSG': str\n",
      "    'ENST': str\n",
      "    'major_consequence': str\n",
      "    'hgvsc': str\n",
      "    'hgvs': str\n",
      "    'category': str\n",
      "    'canonical': int32\n",
      "    'polyphen_pred': str\n",
      "    'sift_pred': str\n",
      "    'most_severe_consequence': str\n",
      "    'variant_class': str\n",
      "    'gene_id_array': array<str>\n",
      "    'gene_symbol_array': array<str>\n",
      "    'variant_qc': struct {\n",
      "        dp_stats: struct {\n",
      "            mean: float64, \n",
      "            stdev: float64, \n",
      "            min: float64, \n",
      "            max: float64\n",
      "        }, \n",
      "        gq_stats: struct {\n",
      "            mean: float64, \n",
      "            stdev: float64, \n",
      "            min: float64, \n",
      "            max: float64\n",
      "        }, \n",
      "        AC: array<int32>, \n",
      "        AF: array<float64>, \n",
      "        AN: int32, \n",
      "        homozygote_count: array<int32>, \n",
      "        call_rate: float64, \n",
      "        n_called: int64, \n",
      "        n_not_called: int64, \n",
      "        n_filtered: int64, \n",
      "        n_het: int64, \n",
      "        n_non_ref: int64, \n",
      "        het_freq_hwe: float64, \n",
      "        p_value_hwe: float64\n",
      "    }\n",
      "    'gnomad_genomes_af': float64\n",
      "    'gnomad_exomes_af': float64\n",
      "    'primate_ai': float64\n",
      "    'splice_ai': float32\n",
      "    'meta_svm': str\n",
      "    'cadd': float32\n",
      "    'gnomad_af': float64\n",
      "    'AC_adj': int32\n",
      "    'AC_adj_Barlow': int32\n",
      "    'AC_adj_FED': int32\n",
      "----------------------------------------\n",
      "Entry fields:\n",
      "    'AD': array<int32>\n",
      "    'DP': int32\n",
      "    'GQ': int32\n",
      "    'GT': call\n",
      "    'MIN_DP': int32\n",
      "    'PGT': call\n",
      "    'PID': str\n",
      "    'PL': array<int32>\n",
      "    'PS': int32\n",
      "    'RGQ': int32\n",
      "    'SB': array<int32>\n",
      "    'adj': bool\n",
      "    'phenotype': str\n",
      "----------------------------------------\n",
      "Column key: ['s']\n",
      "Row key: ['locus', 'alleles']\n",
      "----------------------------------------\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-09-23 15:54:09 Hail: INFO: Reading table without type imputation\n",
      "  Loading field 'Adult-ID' as type str (not specified)\n",
      "  Loading field 'Age_years' as type str (not specified)\n",
      "  Loading field 'group' as type str (not specified)\n",
      "2021-09-23 15:54:09 Hail: WARN: 'GroupedMatrixTable.result': No entry fields were defined.\n"
     ]
    }
   ],
   "source": [
    "# sample information \n",
    "file_input = 'sample_groups.txt'\n",
    "print('Reads in the following txt as a hail table: \\n' + file_input + '\\n')\n",
    "ht = hl.import_table(file_input).key_by('Adult-ID')\n",
    "ht = ht.rename({'Adult-ID' : 's'}) #to match mt\n",
    "\n",
    "# add phenoype group as column annotation\n",
    "mt = mt.annotate_entries(phenotype = ht[mt.s].group)\n",
    "mt.describe()\n",
    "\n",
    "\n",
    "# annotate with AC_adj for each phenotype group\n",
    "mt = mt.annotate_rows(AC_adj_Barlow = hl.agg.filter((mt.phenotype == \"Barlow\") & mt.adj, hl.agg.call_stats(mt.GT, mt.alleles).AC[1]))\n",
    "mt = mt.annotate_rows(AC_adj_FED = hl.agg.filter((mt.phenotype == \"FED\") & mt.adj, hl.agg.call_stats(mt.GT, mt.alleles).AC[1]))\n",
    "\n",
    "\n",
    "\n",
    "misD_mt = mt.filter_rows((mt.gnomad_af <= 0.001) &\n",
    "                             (mt.variant_qc.gq_stats.mean > 98) &\n",
    "                             (mt.AC_adj < 10) &\n",
    "                             (mt.category == \"missense\") &\n",
    "                            (mt.meta_svm == \"D\"))\n",
    "\n",
    "misD_ht = (misD_mt.group_rows_by(misD_mt.gene_symbol)\n",
    "                          .aggregate_rows(gene_count = hl.agg.sum(misD_mt.AC_adj))\n",
    "                          .result()).rows()\n",
    "\n",
    "#misD_ht.show()\n",
    "#misD_ht.export('/gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/misD_gene_counts.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "657cbd2f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-09-23 15:54:24 Hail: WARN: 'GroupedMatrixTable.result': No entry fields were defined.\n",
      "2021-09-23 15:54:31 Hail: INFO: Ordering unsorted dataset with network shuffle0]\n",
      "2021-09-23 15:54:31 Hail: INFO: merging 30 files totalling 26.9K...\n",
      "2021-09-23 15:54:31 Hail: INFO: while writing:\n",
      "    /gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/misD_BD_gene_counts.tsv\n",
      "  merge time: 16.866ms\n",
      "2021-09-23 15:54:32 Hail: WARN: 'GroupedMatrixTable.result': No entry fields were defined.\n",
      "2021-09-23 15:54:34 Hail: INFO: Ordering unsorted dataset with network shuffle0]\n",
      "2021-09-23 15:54:34 Hail: INFO: merging 30 files totalling 26.9K...\n",
      "2021-09-23 15:54:34 Hail: INFO: while writing:\n",
      "    /gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/misD_FED_gene_counts.tsv\n",
      "  merge time: 13.577ms\n"
     ]
    }
   ],
   "source": [
    "\n",
    "misD_BD_ht = (misD_mt.group_rows_by(misD_mt.gene_symbol)\n",
    "                          .aggregate_rows(gene_count = hl.agg.sum(misD_mt.AC_adj_Barlow))\n",
    "                          .result()).rows()\n",
    "\n",
    "#misD_ht.show()\n",
    "\n",
    "misD_BD_ht.export('/gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/misD_BD_gene_counts.tsv')\n",
    "\n",
    "\n",
    "\n",
    "misD_FED_ht = (misD_mt.group_rows_by(misD_mt.gene_symbol)\n",
    "                          .aggregate_rows(gene_count = hl.agg.sum(misD_mt.AC_adj_FED))\n",
    "                          .result()).rows()\n",
    "\n",
    "\n",
    "misD_FED_ht.export('/gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/misD_FED_gene_counts.tsv')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "1cd558f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-09-20 14:23:19 Hail: WARN: 'GroupedMatrixTable.result': No entry fields were defined.\n",
      "2021-09-20 14:23:19 Hail: INFO: Ordering unsorted dataset with network shuffle\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table><thead><tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \" colspan=\"1\"><div style=\"text-align: left;\"></div></td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \" colspan=\"1\"><div style=\"text-align: left;\"></div></td></tr><tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \" colspan=\"1\"><div style=\"text-align: left;border-bottom: solid 2px #000; padding-bottom: 5px\">gene_symbol</div></td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \" colspan=\"1\"><div style=\"text-align: left;border-bottom: solid 2px #000; padding-bottom: 5px\">gene_count</div></td></tr><tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; text-align: left;\">str</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; text-align: left;\">int64</td></tr>\n",
       "</thead><tbody><tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;A2ML1&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">4</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;AARSD1&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;AATF&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;AATK&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;ABC13-47656900H15.1&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">2</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;ABCA1&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;ABCA12&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;ABCA13&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;ABCA4&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "<tr><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">&quot;ABCA7&quot;</td><td style=\"white-space: nowrap; max-width: 500px; overflow: hidden; text-overflow: ellipsis; \">1</td></tr>\n",
       "</tbody></table><p style=\"background: #fdd; padding: 0.4em;\">showing top 10 rows</p>\n"
      ],
      "text/plain": [
       "+-----------------------+------------+\n",
       "| gene_symbol           | gene_count |\n",
       "+-----------------------+------------+\n",
       "| str                   |      int64 |\n",
       "+-----------------------+------------+\n",
       "| \"A2ML1\"               |          4 |\n",
       "| \"AARSD1\"              |          1 |\n",
       "| \"AATF\"                |          1 |\n",
       "| \"AATK\"                |          1 |\n",
       "| \"ABC13-47656900H15.1\" |          2 |\n",
       "| \"ABCA1\"               |          1 |\n",
       "| \"ABCA12\"              |          1 |\n",
       "| \"ABCA13\"              |          1 |\n",
       "| \"ABCA4\"               |          1 |\n",
       "| \"ABCA7\"               |          1 |\n",
       "+-----------------------+------------+\n",
       "showing top 10 rows"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "lof_mt = mt.filter_rows((mt.gnomad_af <= 0.001) &\n",
    "                             (mt.AC_adj < 5) &\n",
    "                             (mt.category == \"lof\") &\n",
    "                             (mt.variant_class == \"SNV\"))\n",
    "\n",
    "'''\n",
    "lof_mt = mt.filter_rows((mt.gnomad_af <= 0.001) &\n",
    "                             (mt.AC_adj < 5) &\n",
    "                             (mt.category == \"lof\"))\n",
    "\n",
    "'''\n",
    "\n",
    "lof_ht = (lof_mt.group_rows_by(lof_mt.gene_symbol)\n",
    "                          .aggregate_rows(gene_count = hl.agg.sum(lof_mt.AC_adj))\n",
    "                          .result()).rows()\n",
    "\n",
    "lof_ht.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "422b14d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-09-20 14:23:25 Hail: INFO: Ordering unsorted dataset with network shuffle\n",
      "2021-09-20 14:23:25 Hail: INFO: merging 30 files totalling 18.7K...\n",
      "2021-09-20 14:23:25 Hail: INFO: while writing:\n",
      "    /gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/lof_gene_counts.tsv\n",
      "  merge time: 29.797ms\n"
     ]
    }
   ],
   "source": [
    "lof_ht.export('/gpfs/ycga/scratch60/ysm/lek/ml2529/gruber_analysis/lof_gene_counts.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7c37f50",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
